tests/decomposition/test_decomposition_graph.py

```diff
@@ -367,7 +367,27 @@ def test_adjoint_adjoint(self, _):
         assert q.queue == [qml.RX(0.5, wires=[0])]
         assert graph.resource_estimate(op) == to_resources({qml.RX: 1})
 
-    def test_custom_adjoint(self, _):
+    def test_adjoint_pow(self, _):
+        """tests that an adjoint of a power of am operator that has adjoint is decomposed."""
+
+        op = qml.adjoint(qml.pow(qml.H(0), z=3))
+
+        graph = DecompositionGraph(operations=[op], target_gate_set={"Hadamard"})
+        # 3 operator nodes: Adjoint(Pow(H)), Pow(H), and H
+        # 2 decomposition nodes for Adjoint(Pow(H)) and Pow(H)
+        assert len(graph._graph.nodes()) == 5
+        # 2 edges from decompositions to ops and 2 edges from ops to decompositions
+        assert len(graph._graph.edges()) == 4
+
+        graph.solve()
+        with qml.queuing.AnnotatedQueue() as q:
+            graph.decomposition(op)(*op.parameters, wires=op.wires, **op.hyperparameters)
+
+        assert q.queue == [qml.pow(qml.H(0), z=3)]
+        # TODO: There should just be a single `H` after we have full support of Pow decompositions.
```

Comment:
What does full support mean here? :) Is there a concrete plan/epic?

Reply:
It's things like `H^2` cancelling out, `X^(1/2)` becoming `SX`, etc.